### PR TITLE
Feature/group update

### PR DESCRIPTION
**Pull request by minevala**
### Motivation
- Users are able to copy resource opening data into other resources

### Description
- Add resource-copy-fieldset into resourcepage

### Tests
- component tests
- simple e2e-tests

Screenshot 2022-02-02 at 16 31 34

### CODE REVIEW

File: azure/master-to-test.pipeline.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # https://aka.ms/yaml
 
 trigger:
-- master
+- feature/group-update
```
Comment (VilleMiekkoja) [inline azure/master-to-test.pipeline.yml:5]: Hmm.. Do we want to revert this back to master after review is done? Anyways I think it was a good way to get it to higher environments as a demo.

Comment (minevala) [reply azure/master-to-test.pipeline.yml:5]: Yes, we do 👍

File: src/components/button/Button.tsx [hunk 2]
```diff
@@ -43,15 +46,23 @@ export function PrimaryButton({
...
+      className={`button-common primary-button ${
+        disabled && 'primary-button--is-disabled'
+      } ${isLoading && 'primary-button--is-loading'} ${className}`}
       variant="primary"
       onClick={onClick}
       type={type}
-      iconLeft={iconLeft}>
+      iconLeft={iconLeft}
+      disabled={disabled}
+      isLoading={isLoading}
```
Comment (VilleMiekkoja) [inline src/components/button/Button.tsx:64]: Nice use of HDS loading button 👍

**Comment by VilleMiekkoja** [comment]
Looking good! Needs maybe some e2e and unit tests.

File: src/resource/page/ResourcePeriodsCopyFieldset.test.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+const testCopyResourceData: TargetResourcesProps = {
+  mainResourceId: 1111,
+  mainResourceName: 'testTargetResource',
+  resources: ['tprek: 1122'],
+};
+
+const onChange = jest.fn();
+
+describe.only(`<ResourcePeriodsCopyFieldset/>`, () => {
```
Comment (minevala) [inline src/resource/page/ResourcePeriodsCopyFieldset.test.tsx:18]: Remember to remove ”.only”

File: src/resource/page/ResourcePeriodsCopyFieldset.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    }
+  };
+
+  return (
+    <div className="resource-copy-date-periods">
+      <Notification
+        type="alert"
+        label={`Olet valinnut joukkopäivityksessä ${
+          (targetResources?.length || 0) + 1
+        } pistettä. Klikkasit "${mainResourceName}"n aukiolotietoa. Sinulle on auennut ”${mainResourceName}”n aukiolotieto muokattavaksi.`}>
```
Comment (VilleMiekkoja) [inline src/resource/page/ResourcePeriodsCopyFieldset.tsx:74]: This sounds pretty TPREK-specific. Are we planning to make the feature tprek only? I guess we can update later if need to.

File: src/App.tsx [hunk 5]
```diff
@@ -38,6 +39,16 @@ const getPersistentTokens = (): OptionalAuthTokens => {
 };
 
 export default function App(): JSX.Element {
+  const hasReferrer = !!document.referrer;
+
+  const searchParams: SearchParameters = urlUtils.parseSearchParameters(
+    window.location.search
```
Comment (VilleMiekkoja) [inline src/App.tsx:45]: Minor comment: we could add window check to avoid crash if in the future hauki goes SSR

**Review by VilleMiekkoja — APPROVED** [review]
Nice work 👍

**Pull request merged**